tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/ILogger/DirectSubmission/Formatting/OtlpLogEventBuilder.cs

```diff
@@ -0,0 +1,106 @@
+using System.Collections.Generic;
+using Datadog.Trace.OpenTelemetry.Logs;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.ILogger.DirectSubmission.Formatting
+{
+    /// <summary>
+    /// Helper class to build OTLP log events from ILogger log calls.
+    /// Extracts structured data and trace context for OTLP export.
+    /// </summary>
+    internal static class OtlpLogEventBuilder
+    {
+        /// <summary>
+        /// Creates a LoggerDirectSubmissionLogEvent with OTLP structured data from ILogger state.
+        /// </summary>
+        public static LoggerDirectSubmissionLogEvent CreateLogEvent<TState>(
+            int logLevel,
+            string categoryName,
+            object eventId,
+            TState state,
+            Exception? exception,
+            Func<TState, Exception?, string> formatter)
+        {
+            var message = formatter(state, exception);
+            var attributes = ExtractAttributes(state, eventId);
+            var (traceId, spanId, flags) = ExtractTraceContext();
+
+            return new LoggerDirectSubmissionLogEvent(null)
+            {
+                OtlpLog = new LogPoint
+                {
+                    Message = message,
+                    LogLevel = logLevel,
+                    CategoryName = categoryName,
+                    Timestamp = DateTimeOffset.UtcNow.DateTime,
+                    Exception = exception,
+                    Attributes = attributes,
+                    TraceId = traceId,
+                    SpanId = spanId,
+                    Flags = flags
+                }
+            };
+        }
+
+        private static Dictionary<string, object?> ExtractAttributes<TState>(TState state, object eventId)
+        {
+            var attributes = new Dictionary<string, object?>();
+
+            // Extract structured properties from ILogger state
+            if (state is IReadOnlyList<KeyValuePair<string, object?>> properties)
+            {
+                foreach (var property in properties)
+                {
+                    if (!string.IsNullOrEmpty(property.Key) && property.Key != "{OriginalFormat}")
+                    {
+                        attributes[property.Key] = property.Value;
+                    }
+                }
+            }
+
+            // Add EventId if present
+            if (eventId.GetHashCode() != 0)
+            {
+                attributes["EventId"] = eventId.GetHashCode();
+            }
+
+            return attributes;
+        }
+
+        private static (TraceId TraceId, ulong SpanId, int Flags) ExtractTraceContext()
+        {
+            // Prefer Datadog's trace context for accurate correlation with DD spans
+            // Fallback to Activity if Datadog tracing is not active
+            var ddSpan = Tracer.Instance.ActiveScope?.Span as Span;
+            if (ddSpan != null)
+            {
+                return (ddSpan.TraceId128, ddSpan.SpanId, 1);
```

Comment:
The flags that must be returned are trace context flags. Currently, the only bit that is expected is the last bit, which says whether the trace has been sampled or not. This should be roughly the calculation, with some additional using statements

```suggestion
                return (ddSpan.TraceId128, ddSpan.SpanId, SamplingPriorityValues.IsKeep(ddSpan.Context.SamplingPriority) ? 1 : 0);
```